internal/controller/operator/factory/build/cache.go

```diff
@@ -0,0 +1,538 @@
+package build
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+	"unicode"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+)
+
+type BasicAuthCreds struct {
+	Username string
+	Password string
+}
+
+type TLSCreds struct {
+	CAFile   string
+	CertFile string
+	KeyFile  string
+	Key      string
+}
+
+func (c *TLSCreds) buildAssetKey(prefix string, name string, key string) string {
+	if name == "" || key == "" {
+		return ""
+	}
+	return fmt.Sprintf("%s_%s_%s", prefix, name, key)
+}
+
+type OAuth2Creds struct {
+	ClientSecret string
+	ClientID     string
+}
+
+type HTTPClientCreds struct {
+	BasicAuth   *BasicAuthCreds
+	BearerToken string
+	OAuth2      *OAuth2Creds
+}
+
+// AssetsCache is a shared cache for all CR assets
+type AssetsCache struct {
```

Comment:
This change doesn't look relevant to this PR. Please remove `cache.go` file.

Reply:
TLSConfig covers only TLS certs, AssetsCache instead covers, BearerTokens, BasicCreds and OAuth2Secrets as well. It allows to remove TLSConfig and also [this](https://github.com/VictoriaMetrics/operator/blob/d6b6873d8a735fdec30d811c3da791437afeba27/internal/controller/operator/factory/vmalert/vmalert.go#L558), [this](https://github.com/VictoriaMetrics/operator/blob/d6b6873d8a735fdec30d811c3da791437afeba27/internal/controller/operator/factory/vmagent/vmagent_scrapeconfig.go#L43) structures and [this](https://github.com/VictoriaMetrics/operator/blob/d6b6873d8a735fdec30d811c3da791437afeba27/internal/controller/operator/factory/k8stools/secret_content.go) file